### PR TITLE
fix(schedule): schedule cannot be disabled if project is archived

### PR DESCRIPTION
**Pull request by guowl3**
#### What this PR does / why we need it:
Added validation to prevent archiving the project if any enabled schedules exist.

#### Which issue(s) this PR fixes:

Fixes #[2180](https://github.com/oceanbase/odc/issues/2180)

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/collaboration/project/ProjectService.java [hunk 1]
```diff
@@ -256,6 +260,9 @@ public Project setArchived(Long id, @NotNull SetArchivedReq req) throws Interrup
         if (!req.getArchived()) {
             throw new BadRequestException("currently not allowed to recover projects");
         }
+        if (scheduleService.hasEnabledScheduleInProject(id)) {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/collaboration/project/ProjectService.java:263]: may also audo disable for shedule while related project archived.

Comment (guowl3) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/collaboration/project/ProjectService.java:263]: check the project status during the next scheduling

**Review by yizhouxw — APPROVED** [review]
lgtm

**Review by MarkPotato777 — APPROVED** [review]
LGTM

**Review by smallsheeeep — APPROVED** [review]
LGTM

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**